pmd-core/src/main/java/net/sourceforge/pmd/util/OptionalBool.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.util;
+
+/** Represents a boolean that may not be present. Use as a non-null type. */
+public enum OptionalBool {
```

Comment:
Should this be public? It sounds like an implementation helper....

Hm... seems to be used in the shadow chain API. Any reason, why you didn't use `Optional<Boolean>` instead?

Reply:
> Hm... seems to be used in the shadow chain API. 

It's also used in some parts of the upcoming typeres API too. It's a common pattern I think, one that is sometimes replaced with a nullable boxed `Boolean`, which I think is... meh


> Any reason, why you didn't use Optional<Boolean> instead?

All three values are known and there's no need to create a new optional every time